### PR TITLE
add generic 'this plugin' if no $name passed

### PR DESCRIPTION
**Pull request by afragen**
create method to accept plugin name

### CODE REVIEW

**Comment by afragen** [comment]
@coenjacobs hold on this. I think I have a better idea. Will update when I get out of operating room and back to computer.

**Comment by afragen** [comment]
My better idea wasn't better. Let me know what you think of this?

File: src/WPUpdatePhp.php [hunk 1]
```diff
@@ -27,7 +27,21 @@ public function __construct( $minimum_version, $recommended_version = null ) {
 	 * @param $name string Name of the plugin to be used in admin notices
 	 */
 	public function set_plugin_name( $name ) {
-		$this->plugin_name = $name;
+		if ( empty( $this->plugin_name ) ) {
```
Comment (coenjacobs) [inline src/WPUpdatePhp.php:30]: I think this if-statement needs to be the other way around? Either way, do you think this is a good case, as the parameter is required and if you purposefully try to break it by passing `NULL` or anything like that, I'm fine with it displaying weird in the admin notice. Don't you agree?

File: src/WPUpdatePhp.php [hunk 2]
```diff
@@ -27,7 +27,21 @@ public function __construct( $minimum_version, $recommended_version = null ) {
...
+			$this->plugin_name = $name;
+		}
+	}
+
+	/**
+	 * @param $name
+	 *
+	 * @return bool
+	 */
+	public function does_my_plugin_meet_required_php_version( $name ) {
```
Comment (coenjacobs) [inline src/WPUpdatePhp.php:42]: Would you use this function to do the name set and version check in one call? Just syntactic sugar, right (just making sure I don't miss anything)?

**Comment by afragen** [comment]
Much of what I'm trying to figure out here is how to create translatable strings and this would include "this plugin". That's why I originally had that as the default for `$plugin_name` though that won't work as it's not translatable in that location.

We could add `$this->plugin_name = __( 'this plugin' );` to the constructor and then the name set function would be a lot simpler. Then the admin notices could have default or actual plugin names.

If you want me to adjust this PR or create a new one to show you let me know. My goal is to make the strings i18n.

File: src/WPUpdatePhp.php (hunk 2 above)
Comment (afragen) [inline src/WPUpdatePhp.php:42]: Yes, that is the purpose of this is function. Calling `$updatePhp->set_plugin_name` Before the version check is a possibility, but we have to ensure that the plugin name or the default generic string is available to the admin notices if the name set isn't called. Make sense?

**Comment by afragen** [comment]
I think #27 is a better construct. Set `$plugin_name` in constructor and refactor notices.

**Pull request closed without merging**